### PR TITLE
feat: [PL-42038]: Organization Event Publisher in Redis

### PR DESCRIPTION
**Pull request by adiyaar24**
## Describe your changes
Enhance organization event handling to include scopeInfo and uniqueId of organization
Tested scenario: 
- Create Org event published correctly
- Delete Org event published correctly

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57111)

### CODE REVIEW

File: 953-events-api/src/main/proto/io/harness/eventsframework/schemas/entity_crud/organization/organization.proto [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 package io.harness.eventsframework.entity_crud.organization;
-
 option java_multiple_files = true;
 option go_package = "953_events_api";
 
 message OrganizationEntityChangeDTO {
   string accountIdentifier = 1;
   string identifier = 2;
+  string uniqueId = 3;
+  io.harness.eventsframework.schemas.entity.ScopeInfo scopeinfp = 4;
```
Comment (ankit-harness) [inline 953-events-api/src/main/proto/io/harness/eventsframework/schemas/entity_crud/organization/organization.proto:17]: spelling mistake

File: 970-ng-commons/src/main/proto/io/harness/eventsframework/schemas/entity/scope_info.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+syntax = "proto3";
+
+package io.harness.eventsframework.schemas.entity;
+
+option java_multiple_files = true;
+option go_package = "970_ng_commons";
+
+message ScopeInfo {
+  string uniqueId = 1;
+  string parentUniqueId = 2;
```
Comment (ankit-harness) [inline 970-ng-commons/src/main/proto/io/harness/eventsframework/schemas/entity/scope_info.proto:15]: Why parentUniqueId is needed?

File: 970-ng-commons/src/main/proto/io/harness/eventsframework/schemas/entity/scope_info.proto [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  string scopeLevel = 3;
```
Comment (ankit-harness) [inline 970-ng-commons/src/main/proto/io/harness/eventsframework/schemas/entity/scope_info.proto:16]: Should this be enum?
Can we rename it to scope?

**Review by ankit-harness — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57111)* status: 5 file(s) to review. 0 of 2 approvals obtained.

**Comment by pbarapatre10** [comment]
trigger uts

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57111#-:-NnD5ytiBgQzYAb5kD0y:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57111#-:-NnD6Mo3BfbWybR-Ni3z:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Comment by pbarapatre10** [comment]
trigger uts

**Comment by pbarapatre10** [comment]
trigger uts

**Comment by pbarapatre10** [comment]
trigger uts

**Comment by pbarapatre10** [comment]
trigger uts

**Comment by pbarapatre10** [comment]
trigger ut7

**Comment by pbarapatre10** [comment]
trigger ut6

**Comment by pbarapatre10** [comment]
trigger ut5

**Comment by pbarapatre10** [comment]
trigger uts

**Comment by pbarapatre10** [comment]
trigger ut9

**Comment by pbarapatre10** [comment]
trigger ut0

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57111#-:-NnRnJvo9USVHnbQWi7M:bnfp4nl)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**